packages/react-youtube/src/YouTube.tsx

```diff
@@ -300,37 +306,60 @@ class YouTube extends React.Component<YouTubeProps> {
    */
   onPlayerPlaybackQualityChange = (event: YouTubeEvent<string>) => this.props.onPlaybackQualityChange?.(event);
 
+  /**
+   * Destroy the YouTube Player using its async API and store the promise so we
+   * can await before re-creating it.
+   */
+  destroyPlayer = () => {
+    if (this.internalPlayer) {
+      this.destroyPlayerPromise = this.internalPlayer.destroy().then(() => (this.destroyPlayerPromise = undefined));
+    }
+  };
+
   /**
    * Initialize the YouTube Player API on the container and attach event handlers
    */
   createPlayer = () => {
-    // do not attempt to create a player server-side, it won't work
-    if (typeof document === 'undefined') return;
-    // create player
-    const playerOpts: Options = {
-      ...this.props.opts,
-      // preload the `videoId` video if one is already given
-      videoId: this.props.videoId,
+    const doCreatePlayer = () => {
+      // do not attempt to create a player server-side, it won't work
+      if (typeof document === 'undefined') return;
+      // create player
```

Comment:
Instead of wrapping the `createPlayer` logic into a function, I suggest leaving as it was before and add the following here:
```suggestion
    if (this.destroyPlayerPromise) {
      // We need to first await the existing player to be destroyed before
      // we can re-create it.
      this.destroyPlayerPromise.then(this.createPlayer);
      return;
    }
    // create player
```